flink-tests/src/test/java/org/apache/flink/test/checkpointing/ChangelogPeriodicMaterializationSwitchStateBackendITCase.java

```diff
@@ -50,6 +86,78 @@ public void testSwitchFromEnablingToDisablingWithRescalingIn() throws Exception
         testSwitchEnv(getEnv(true, NUM_SLOTS), getEnv(false, NUM_SLOTS / 2));
     }
 
+    @Test
+    public void testSwitchFromDisablingToEnablingInClaimMode() throws Exception {
+        File firstCheckpointFolder = TEMPORARY_FOLDER.newFolder();
+        MiniCluster miniCluster = cluster.getMiniCluster();
+        StreamExecutionEnvironment env1 =
+                getEnv(delegatedStateBackend, firstCheckpointFolder, false, 100, 600000);
```

Comment:
This test relies on materialization, but the interval is quite big and might prevent it from happening - should it be decreased?

And ideally, we should verify the presense of the materialized part - similar to `ChangelogPeriodicMaterializationITCase`.

Reply:
This test scenario is:
  1. changelog disabled(the interval of materialization doesn't work here), get chk-1  to restore.
  2. changelog enabled, retore from chk-1, get chk-2(**no materialization occurs, the materialized part of chk-2 is in chk-1**).
  3. changelog enable, restore from chk-2, test whether the materialized part in chk-1 is exist.

So, the interval is deliberately set to be long here to avoid materializing. To simulate a scenario where one JOB restore again before one materialization occurs.

Reply:
I see, thanks for clarifying.